README.md

```diff
@@ -50,3 +50,27 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+Worst-case time:
+Number of vertices/nodes (cities) = n
+Initial tour generation (randomized) = $\Theta$(n) time
+2-OptSwap & Tour length recalculation = $\Theta$(n) time
+worst case iterations = 1000
+
+worst case time = $\Theta(1000 \times n) = \Theta(n)$
+
+worst-case space = $\Theta(n^2)$
+
+
+The worst-case time complexity is $\Theta(n^2)$, where n is the number of cities. This is because he algorithm starts with generating an initial random tour (usually done at setup), which typically takes $\Theta$(n) time. Then, during each iteration, the algorithm does a 2-opt swap and recalculates the tour length, both taking $\Theta$(n) time. With up to 1000 iterations in the worst case, the worst-case time complexity is $\Theta(n^2)$.
```

Comment:
But if you find an improvement, the counter resets. How does that affect the overall complexity?